src/basic/user-util.c

```diff
@@ -148,7 +148,7 @@ static int synthesize_user_creds(
                         *home = FLAGS_SET(flags, USER_CREDS_CLEAN) ? NULL : "/";
 
                 if (shell)
-                        *shell = FLAGS_SET(flags, USER_CREDS_CLEAN) ? NULL : "/sbin/nologin";
+                        *shell = FLAGS_SET(flags, USER_CREDS_CLEAN) ? NULL : NOLOGIN;
```

Comment:
hmm, this is needlessly cryptic. Maybe NOLOGIN_PATH would be a better name, matching the existing MOUNT_PATH and UMOUNT_PATH?

Reply:
We are not consistent here. While we have `MOUNT_PATH` and `UMOUNT_PATH`, we also have `TELINIT`, `SULOGIN`,  `KMOD`, etc without a _PATH suffix. The latter seems more commonly used , which is why I picked the shorter version without the path. I can change it to `NOLOGIN_PATH` though if you prefer.

Reply:
That's because there's already MOUNT macro (for casting units). I think NOLOGIN is appropriate. (Or we should change all of them to have the `_PATH` suffix.)

Reply:
So the thing that trips me up is that there's also /etc/nologin + /run/nologin, and it's awfully confusing to have all this and then be so short that we just have NOLOGIN?

Maybe call this NOLOGIN_BINARY to make clear it's not about those other files we also manage?

Reply:
Which specific part of the code becomes hard to read? Not quite sure which place you mean with /etc/nologin + /run/nologin, tbh.

Reply:
do git grep "nologin" in our sources and you'll see nologin_timeout_handler() and related code in logind and tmpfiles that handles a /run/nologin file that has almost nothing to do with the nologin binary... It's UNIX fault I guess, but I think we should probably work towards keeping them as separete as we can, and in particular for global defines avoid just referring to anything as just "NOLOGIN", since it's not obvious what that is: it could be the path to /run/nologin or the path to /usr/sbin/nologin, the name doesn't help you at all, and we refer to both paths in our codebase. Hence, let's avoid the confusion, and work towards cleaning this up at least a bit, by naming this #defined NOLOGIN_BINARY or so, which make this very clear to grok

Reply:
(i mean, let's not forget that /run/nologin also moved from one path to another, so it's very similar territory, so the uninitiated reader might very well assume that the NOLOGIN macro refers to that variable path, even though it refers to a very different concept and variable path)

Reply:
If we are not going to use an existing naming scheme for this variable, maybe `NOLOGIN_SHELL` would be best to indicate what it is about?